lib/socks.c

```diff
@@ -782,13 +783,9 @@ CURLcode Curl_SOCKS5(const char *proxy_user,
       return CURLE_COULDNT_RESOLVE_HOST;
     }
 
-    if(Curl_printable_address(hp, dest, sizeof(dest))) {
-      size_t destlen = strlen(dest);
-      msnprintf(dest + destlen, sizeof(dest) - destlen, ":%d", remote_port);
-    }
-    else {
-      strcpy(dest, "unknown");
-    }
+    Curl_printable_address(hp, dest, sizeof(dest));
+    destlen = strlen(dest);
+    msnprintf(dest + destlen, sizeof(dest) - destlen, ":%d", remote_port);
```

Comment:
Was it intended that this could end up with unknown:123

Reply:
I don't understand. Before this change there was a theoretical chance it could put `unknown`  (without port number) into the buffer. Now, if Curl_printable_address fails, this logic will instead put `:[portnumber]` in the buffer. Ie a failed printable address will just be blank. I did this partly to simplify but also because it is virtually unheard of that Curl_printable_address failed...

Reply:
If dest is initialized to "unknown" and if Curl_printable_address fails then it is unknown:123. The old way was it would just say unknown.

> I did this partly to simplify but also because it is virtually unheard of that Curl_printable_address failed...

Sure I agree, I just wanted to know if that's what you intended.